app/src/services/bankService.ts

```diff
@@ -0,0 +1,75 @@
+import { EthersJsAdapter, type IWeb3Library } from '@/adapters/web3LibraryAdapter'
+import { FetchTeamAPI, type TeamAPI } from '@/apis/teamApi'
+import BANK_ABI from '../artifacts/abi/bank.json'
+import type { Contract } from 'ethers'
+
+export interface IBankService {
+  web3Library: IWeb3Library
+  teamApi: TeamAPI
+  createBankContract(id: string): Promise<string>
+  deposit(bankAddress: string, amount: string): Promise<any>
+  pushTip(bankAddress: string, addresses: string[], amount: number): Promise<any>
+  sendTip(bankAddress: string, addresses: string[], amount: number): Promise<any>
+}
+
+export class BankService implements IBankService {
+  web3Library: IWeb3Library
+  teamApi: TeamAPI
```

Comment:
why is this needed here?

Reply:
- web3Library: because i need it to do some of web3 stuff in the service
- teamApi: i use it because i need to update the bankAddress to deployed bank contract address using update endpoint